src/folding/nova/nifs.rs

```diff
@@ -85,36 +85,49 @@ where
         CommittedInstance::<C> { cmE, u, cmW, x }
     }
 
-    // NIFS.P
-    #[allow(clippy::type_complexity)]
-    pub fn prove(
+    /// NIFS.P is the consecutive combination of compute_cmT with fold_instances
+
+    /// compute_cmT is part of the NIFS.P logic
+    pub fn compute_cmT(
         pedersen_params: &PedersenParams<C>,
-        // r comes from the transcript, and is a n-bit (N_BITS_CHALLENGE) element
-        r: C::ScalarField,
         r1cs: &R1CS<C::ScalarField>,
         w1: &Witness<C>,
         ci1: &CommittedInstance<C>,
         w2: &Witness<C>,
         ci2: &CommittedInstance<C>,
-    ) -> Result<(Witness<C>, CommittedInstance<C>, Vec<C::ScalarField>, C), Error> {
+    ) -> Result<(Vec<C::ScalarField>, C), Error> {
         let z1: Vec<C::ScalarField> = [vec![ci1.u], ci1.x.to_vec(), w1.W.to_vec()].concat();
         let z2: Vec<C::ScalarField> = [vec![ci2.u], ci2.x.to_vec(), w2.W.to_vec()].concat();
 
         // compute cross terms
         let T = Self::compute_T(r1cs, ci1.u, ci2.u, &z1, &z2)?;
-        let rT = C::ScalarField::one(); // use 1 as rT since we don't need hiding property for cm(T)
-        let cmT = Pedersen::commit(pedersen_params, &T, &rT)?;
+        // use r_T=1 since we don't need hiding property for cm(T)
+        let cmT = Pedersen::commit(pedersen_params, &T, &C::ScalarField::one())?;
+        Ok((T, cmT))
+    }
 
+    /// fold_instances is part of the NIFS.P logic
+    pub fn fold_instances(
+        // r comes from the transcript, and is a n-bit (N_BITS_CHALLENGE) element
+        r: C::ScalarField,
+        w1: &Witness<C>,
+        ci1: &CommittedInstance<C>,
+        w2: &Witness<C>,
+        ci2: &CommittedInstance<C>,
```

Comment:
All these impl `Copy` no? Why not sending them without the ref? They're going to be copied anyways.

Reply:
Couldn't implement `Copy` because it contains a vector (
https://github.com/privacy-scaling-explorations/folding-schemes/tree/247c6d63cdf2b6f4e4551e2f3b0ac35c6aedf003/src/folding/nova/mod.rs#L25 ). Should we define the structs differently to allow it?

Reply:
Ohh I did not chech the internal typing thouugh.

Nono, it's fine as is! :) Thxz and sorry for the mess on this comment